src/NSubstitute/Core/Argument.cs

```diff
@@ -26,9 +26,9 @@ public Argument(ICall call, int argIndex)
             _argIndex = argIndex;
         }
 
-        public object Value
+        public object? Value
         {
-            get => _getValue != null ? _getValue() : _call.GetArguments()[_argIndex];
+            get => _getValue != null ? _getValue() : _call!.GetArguments()[_argIndex];
```

Comment:
**question:** is there a reason to use `!.` rather than `?.` here?

Reply:
We don't have discriminated unions in C#, so this class implements them via logic. I definitely know if `_getValue` is null, then `_call` is always present. The invariants are guaranteed by the constructors.

The `_call?.GetArguments()` would have different semantics (we accept that the field _could_ be nullable) and different result type (`object?` vs `object`, even though in this case it does not matter).